asmrun/signals_asm.c

```diff
@@ -214,38 +214,69 @@ DECLARE_SIGNAL_HANDLER(segv_handler)
   struct rlimit limit;
   struct sigaction act;
   char * fault_addr;
+  int handled = 0;
 
   /* Sanity checks:
      - faulting address is word-aligned
      - faulting address is within the stack
      - we are in OCaml code */
   fault_addr = CONTEXT_FAULTING_ADDRESS;
   if (((uintnat) fault_addr & (sizeof(intnat) - 1)) == 0
-      && getrlimit(RLIMIT_STACK, &limit) == 0
-      && fault_addr < system_stack_top
-      && fault_addr >= system_stack_top - limit.rlim_cur - EXTRA_STACK
+      && fault_addr < caml_top_of_stack) {
+    size_t stack_size;
+
```

Comment:
Even though that's not what the original code did, I would prefer to have the code that computes the stack size first, followed by the "Sanity checks" tests, e.g.:
```
   /* Compute the size of the stack for the thread that segfaulted */
  if (caml_stack_size < 0) ... else if (caml_stack_size == 0) { ... } else { ... }
  /* Sanity checks: bla bla bla */
  if (test1 && test2 && ... && testN) {
```
The sanity checks succeed most of the time, so there is no efficiency to be gained by conditionalizing the stack size computation.  Plus, this code is not performance-critical.